java/ql/lib/semmle/code/java/security/UnsafeDeserializationQuery.qll

```diff
@@ -224,6 +226,11 @@ predicate unsafeDeserialization(MethodCall ma, Expr sink) {
     m instanceof GsonDeserializeMethod and
     sink = ma.getArgument(0) and
     UnsafeTypeFlow::flowToExpr(ma.getArgument(1))
+    or
+    m.getASourceOverriddenMethod*() instanceof ObjectMessageGetObjectMethod and
+    sink = ma.getQualifier().getUnderlyingExpr() and
+    // If we can see an implementation, we trust dataflow to find a path to the other sinks instead
+    not exists(viableCallable(ma))
```

Comment:
So with this implementation, we're only counting the sink if we can't determine more about the context (via the available implementations of JMS)?

Reply:
Exactly. My reasoning is that, if we see the implementation, it means it belongs to the project and we can analyze data flow through it, so if it's really doing deserialization unsafely, the other sinks will catch it. If we don't see the implementation it means it's in a dependency, in which case we prefer to err on the side of caution and alert.